evm-precompiles/erc721/src/lib.rs

```diff
@@ -274,23 +268,76 @@ where
 			return Err(error("expected token id <= 2^32").into());
 		}
 		let serial_number: SerialNumber = serial_number.saturated_into();
+		let token_id = (series_id_parts.0, series_id_parts.1, serial_number);
+		let approved_account: H160 = Runtime::AddressMapping::from_account_id(
+			crml_token_approvals::Module::<Runtime>::erc721_approvals(token_id),
+		);
 
 		// Build call with origin.
-		if context.caller == from {
-			let from = Runtime::AddressMapping::into_account_id(context.caller);
+		if context.caller == from || context.caller == approved_account {
+			let from = Runtime::AddressMapping::into_account_id(from);
 			let to = Runtime::AddressMapping::into_account_id(to);
 
 			// Dispatch call (if enough gas).
 			RuntimeHelper::<Runtime>::try_dispatch(
 				Some(from).into(),
 				crml_nft::Call::<Runtime>::transfer {
-					token_id: (series_id_parts.0, series_id_parts.1, serial_number),
+					token_id,
 					new_owner: to,
 				},
 				gasometer,
 			)?;
 		} else {
-			// TODO: use approval stuff..
+			return Err(error("caller not approved").into());
+		}
+
+		// Build output.
+		Ok(PrecompileOutput {
+			exit_status: ExitSucceed::Returned,
+			cost: gasometer.used_gas(),
+			output: EvmDataWriter::new().write(true).build(),
+			logs: Default::default(),
```

Comment:
need to add the `Transfer` log for this operation to match erc721 and allow dapps to distinguish between normal transfer and approval transfer